Test regression on PyPy3.9

Fixtures 4.0.0 fails tests on PyPy3.9 7.3.9:

```
pypy3 run-test: commands[0] | make check
python -m testtools.run fixtures.test_suite
/usr/lib/pypy3.9/runpy.py:127: RuntimeWarning: 'testtools.run' found in sys.modules after import of package 'testtools', but prior to execution of 'testtools.run'; this may result in unpredictable behaviour
  warn(RuntimeWarning(msg))
Tests running...
======================================================================
ERROR: fixtures.tests._fixtures.test_monkeypatch.TestMonkeyPatch.test_patch_classmethod_with_classmethod
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/fixtures/fixtures/tests/_fixtures/test_monkeypatch.py", line 200, in test_patch_classmethod_with_classmethod
    cls, = C.foo_cls()
ValueError: too many values to unpack (expected 1)
======================================================================
FAIL: fixtures.tests._fixtures.test_monkeypatch.TestMonkeyPatch.test_patch_classmethod_with_boundmethod
----------------------------------------------------------------------
Failed expectation: {{{MismatchError: <class 'fixtures.tests._fixtures.test_monkeypatch.C'> is not None}}}
Failed expectation-1: {{{MismatchError: <class 'fixtures.tests._fixtures.test_monkeypatch.C'> is not None}}}

AssertionError: Forced Test Failure

Ran 134 tests in 2.050s
FAILED (failures=2)
```

This seems to be caused by the following commit:
```
commit fe830674abd4926d96d38f9992f3e31b00cd891a (HEAD, refs/bisect/bad)
Author: Stephen Finucane <stephenfin@redhat.com>
Date:   2021-02-25 12:37:42 +0100

    Fix tests on Python 3.9
    
    I'm not entirely sure this is correct, but it's the only thing I can
    find related to changes in classmethod in Python 3.9.
    
    Signed-off-by: Stephen Finucane <stephenfin@redhat.com>
```

--- a/fixtures/tests/_fixtures/test_monkeypatch.py
+++ b/fixtures/tests/_fixtures/test_monkeypatch.py
@@ -23,6 +23,9 @@ from fixtures import MonkeyPatch, TestWithFixtures
 
 reference = 23
 
+NEW_PY39_CLASSMETHOD = (
+    sys.version_info >= (3, 9) and not hasattr(sys, "pypy_version_info"))
+
 class C(object):
     def foo(self, arg):
         return arg
@@ -196,7 +199,7 @@ class TestMonkeyPatch(testtools.TestCase, TestWithFixtures):
             # with the class
             #
             # https://bugs.python.org/issue19072
-            if sys.version_info >= (3, 9):
+            if NEW_PY39_CLASSMETHOD:
                 cls, = C.foo_cls()
                 self.expectThat(cls, Is(D))
                 cls, = C().foo_cls()
@@ -238,13 +241,13 @@ class TestMonkeyPatch(testtools.TestCase, TestWithFixtures):
             self.expectThat(slf, Is(d))
             # See note in test_patch_classmethod_with_classmethod on changes in
             # Python 3.9
-            if sys.version_info >= (3, 9):
+            if NEW_PY39_CLASSMETHOD:
                 self.expectThat(cls, Is(None))
             else:
                 self.expectThat(cls, Is(C))
             slf, cls = C().foo_cls()
             self.expectThat(slf, Is(d))
-            if sys.version_info >= (3, 9):
+            if NEW_PY39_CLASSMETHOD:
                 self.expectThat(cls, Is(None))
             else:
                 self.expectThat(cls, Is(C))

